recipe/build.sh

```diff
@@ -1,44 +1,91 @@
 #!/bin/bash
 
+set -x
 
-for USE_WIDEC in false true;
-do
-    WIDEC_OPT=""
-    if [ "${USE_WIDEC}" = true ];
-    then
-        WIDEC_OPT="--enable-widec"
-    fi
+# ncurses (gen-pkgconfig.in) adds -ltinfo to ncurses.pc if any of the following conditions is true:
+# 1. No `*-Wl,-rpath,*` is found in EXTRA_LDFLAGS
+# 2. Any `*--as-needed*` is found in EXTRA_LDFLAGS
+# The build system takes care that any `-Wl,-rpath,` in LDFLAGS gets copied into EXTRA_LDFLAGS
+# (and also that any -L${PREFIX}/lib gets translated to -Wl,-rpath,${PREFIX}/lib)
+# the same is not true of -Wl,--as-needed (which is referenced only in gen-pkgconfig.in).
+# One option to fix this is to pass our LDFLAGS as EXTRA_LDFLAGS however this ends up copying across
+# all of our linker flags into the .pc file which means they are forced upon all pkg-config based
+# consumers of ncurses and that is a very bad thing indeed. If we wanted to do that we would:
+# export EXTRA_LDFLAGS=${LDFLAGS}
+# export LDFLAGS=
+# .. but instead it is better to strip off all '-Wl,-rpath,*' and '-L${PREFIX}' from LDFLAGS.
```

Comment:
@mingwandroid, in light of the fact that we need rpath for libc++.dylib, is this a good idea?